File: SDKs/amazonaws.com/redshift/2012-12-01/go/pkg/models/shared/clustersubnetgroup.go
package shared

// ClusterSubnetGroup
// Describes a subnet group.
type ClusterSubnetGroup struct {
	ClusterSubnetGroupName *string
	Description            *string
	SubnetGroupStatus      *string
	Subnets                []SubnetList
	Tags                   []TagList
	VpcID                  *string
}

File: SDKs/amazonaws.com/redshift/2012-12-01/typescript/src/sdk/models/shared/clustersubnetgroup.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { SubnetList } from "./subnetlist";
import { TagList } from "./taglist";



// ClusterSubnetGroup
/** 
 * Describes a subnet group.
**/
export class ClusterSubnetGroup extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clusterSubnetGroupName?: string;

  @SpeakeasyMetadata()
  description?: string;

  @SpeakeasyMetadata()
  subnetGroupStatus?: string;

  @SpeakeasyMetadata({ elemType: SubnetList })
  subnets?: SubnetList[];

  @SpeakeasyMetadata({ elemType: TagList })
  tags?: TagList[];

  @SpeakeasyMetadata()
  vpcId?: string;
}
